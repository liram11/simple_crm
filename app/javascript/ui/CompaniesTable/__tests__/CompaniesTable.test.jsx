/**
 * @jest-environment jsdom
 */

import React from "react";
import renderer from "react-test-renderer";
import { CompaniesTable } from "..";

const companies = [
  {
    "id": 10,
    "name": "Wisozk, Kshlerin and McCullough",
    "employee_count": 698,
    "industry": "Philanthropy",
    "created_at": "2024-02-16T09:05:57.179Z",
    "updated_at": "2024-02-16T09:05:57.179Z",
    "deals_sum": 776
  },
  {
    "id": 9,
    "name": "Cronin, Wilkinson and Brown",
    "employee_count": 698,
    "industry": "Museums and Institutions",
    "created_at": "2024-02-16T09:05:57.177Z",
    "updated_at": "2024-02-16T09:05:57.177Z",
    "deals_sum": 1161
  },
  {
    "id": 8,
    "name": "Fay and Sons",
    "employee_count": 89,
    "industry": "Think Tanks",
    "created_at": "2024-02-16T09:05:57.176Z",
    "updated_at": "2024-02-16T09:05:57.176Z",
    "deals_sum": 2819
  },
  {
    "id": 7,
    "name": "Wunsch LLC",
    "employee_count": 238,
    "industry": "Public Policy",
    "created_at": "2024-02-16T09:05:57.174Z",
    "updated_at": "2024-02-16T09:05:57.174Z",
    "deals_sum": 1243
  },
  {
    "id": 6,
    "name": "Gutmann and Sons",
    "employee_count": 48,
    "industry": "Glass, Ceramics & Concrete",
    "created_at": "2024-02-16T09:05:57.167Z",
    "updated_at": "2024-02-16T09:05:57.167Z",
    "deals_sum": 1205
  },
  {
    "id": 5,
    "name": "Hammes, Moen and Purdy",
    "employee_count": 398,
    "industry": "Research",
    "created_at": "2024-02-16T09:05:57.159Z",
    "updated_at": "2024-02-16T09:05:57.159Z",
    "deals_sum": 2298
  },
  {
    "id": 4,
    "name": "Hintz LLC",
    "employee_count": 245,
    "industry": "Public Safety",
    "created_at": "2024-02-16T09:05:57.142Z",
    "updated_at": "2024-02-16T09:05:57.142Z",
    "deals_sum": 730
  },
  {
    "id": 3,
    "name": "Greenholt Group",
    "employee_count": 27,
    "industry": "International Affairs",
    "created_at": "2024-02-16T09:05:57.140Z",
    "updated_at": "2024-02-16T09:05:57.140Z",
    "deals_sum": 2048
  },
  {
    "id": 2,
    "name": "Mohr and Sons",
    "employee_count": 594,
    "industry": "Consumer Services",
    "created_at": "2024-02-16T09:05:57.138Z",
    "updated_at": "2024-02-16T09:05:57.138Z",
    "deals_sum": 1004
  },
  {
    "id": 1,
    "name": "Jacobson-Hamill",
    "employee_count": 881,
    "industry": "Venture Capital & Private Equity",
    "created_at": "2024-02-16T09:05:57.136Z",
    "updated_at": "2024-02-16T09:05:57.136Z",
    "deals_sum": 2246
  }
];

// eslint-disable-next-line no-undef
it("renders correctly", () => {
  const tree = renderer
    .create(<CompaniesTable companies={companies} page={1} totalPages={2} onPageChange={() => {}} />)
    .toJSON();
  // eslint-disable-next-line no-undef
  expect(tree).toMatchSnapshot();
});
