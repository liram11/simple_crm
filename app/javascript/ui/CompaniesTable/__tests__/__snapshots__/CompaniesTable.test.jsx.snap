// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Name
        </th>
        <th
          scope="col"
        >
          Industry
        </th>
        <th
          scope="col"
        >
          Employee Count
        </th>
        <th
          scope="col"
        >
          Total Deal Amount
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Wisozk, Kshlerin and McCullough
        </td>
        <td>
          Philanthropy
        </td>
        <td>
          698
        </td>
        <td>
          776
        </td>
      </tr>
      <tr>
        <td>
          Cronin, Wilkinson and Brown
        </td>
        <td>
          Museums and Institutions
        </td>
        <td>
          698
        </td>
        <td>
          1161
        </td>
      </tr>
      <tr>
        <td>
          Fay and Sons
        </td>
        <td>
          Think Tanks
        </td>
        <td>
          89
        </td>
        <td>
          2819
        </td>
      </tr>
      <tr>
        <td>
          Wunsch LLC
        </td>
        <td>
          Public Policy
        </td>
        <td>
          238
        </td>
        <td>
          1243
        </td>
      </tr>
      <tr>
        <td>
          Gutmann and Sons
        </td>
        <td>
          Glass, Ceramics & Concrete
        </td>
        <td>
          48
        </td>
        <td>
          1205
        </td>
      </tr>
      <tr>
        <td>
          Hammes, Moen and Purdy
        </td>
        <td>
          Research
        </td>
        <td>
          398
        </td>
        <td>
          2298
        </td>
      </tr>
      <tr>
        <td>
          Hintz LLC
        </td>
        <td>
          Public Safety
        </td>
        <td>
          245
        </td>
        <td>
          730
        </td>
      </tr>
      <tr>
        <td>
          Greenholt Group
        </td>
        <td>
          International Affairs
        </td>
        <td>
          27
        </td>
        <td>
          2048
        </td>
      </tr>
      <tr>
        <td>
          Mohr and Sons
        </td>
        <td>
          Consumer Services
        </td>
        <td>
          594
        </td>
        <td>
          1004
        </td>
      </tr>
      <tr>
        <td>
          Jacobson-Hamill
        </td>
        <td>
          Venture Capital & Private Equity
        </td>
        <td>
          881
        </td>
        <td>
          2246
        </td>
      </tr>
    </tbody>
  </table>
  <div>
    <ul
      className="pagination"
    >
      <li
        className="page-item"
      >
        <span
          className="page-link disabled"
        >
          Page: 
          1
        </span>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          Previous Page
        </a>
      </li>
      <li
        className="page-item"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          Next Page
        </a>
      </li>
    </ul>
  </div>
</div>
`;
